pkg/services/alerting/notifiers/alertmanager.go

```diff
@@ -103,41 +110,57 @@ func (this *AlertmanagerNotifier) createAlert(evalContext *alerting.EvalContext,
 	return alertJSON
 }
 
-func (this *AlertmanagerNotifier) Notify(evalContext *alerting.EvalContext) error {
-	this.log.Info("Sending Alertmanager alert", "ruleId", evalContext.Rule.Id, "notification", this.Name)
+// Notify sends alert notifications to the alert manager
+func (am *AlertmanagerNotifier) Notify(evalContext *alerting.EvalContext) error {
+	am.log.Info("Sending Alertmanager alert", "ruleId", evalContext.Rule.Id, "notification", am.Name)
 
-	ruleUrl, err := evalContext.GetRuleUrl()
+	ruleURL, err := evalContext.GetRuleUrl()
 	if err != nil {
-		this.log.Error("Failed get rule link", "error", err)
+		am.log.Error("Failed get rule link", "error", err)
 		return err
 	}
 
 	// Send one alert per matching series.
 	alerts := make([]interface{}, 0)
 	for _, match := range evalContext.EvalMatches {
-		alert := this.createAlert(evalContext, match, ruleUrl)
+		alert := am.createAlert(evalContext, match, ruleURL)
 		alerts = append(alerts, alert)
 	}
 
 	// This happens on ExecutionError or NoData
 	if len(alerts) == 0 {
-		alert := this.createAlert(evalContext, nil, ruleUrl)
+		alert := am.createAlert(evalContext, nil, ruleURL)
 		alerts = append(alerts, alert)
 	}
 
 	bodyJSON := simplejson.NewFromAny(alerts)
 	body, _ := bodyJSON.MarshalJSON()
 
 	cmd := &m.SendWebhookSync{
-		Url:        this.Url + "/api/v1/alerts",
+		Url:        am.URL + "/api/v1/alerts",
 		HttpMethod: "POST",
 		Body:       string(body),
 	}
 
 	if err := bus.DispatchCtx(evalContext.Ctx, cmd); err != nil {
-		this.log.Error("Failed to send alertmanager", "error", err, "alertmanager", this.Name)
+		am.log.Error("Failed to send alertmanager", "error", err, "alertmanager", am.Name)
 		return err
 	}
 
 	return nil
 }
+
+func replaceIllegalCharsInLabelname(input string) string {
+	re := regexp.MustCompile(`[a-zA-Z_][a-zA-Z0-9_]*`)
+
+	var result []byte
+	for _, c := range input {
```

Comment:
this feels quite insufficient.

Reply:
Maybe you can negate instead and use replace all illegal characters which are not any of these with `_`?